src/scripts/main.js

```diff
@@ -1 +1,75 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', () => {
+    resolve('First promise was resolved');
+  });
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+});
```

Comment:
The first promise may encounter an issue where both the click event and the timeout can trigger, leading to an unhandled rejection. Consider adding a flag to check if the promise has already been resolved before calling `reject` in the timeout.